### PR TITLE
Validate node_modules exists when publishing components

### PR DESCRIPTION
**Pull request by eahefnawy**
Otherwise, the Lambda function would return an `internal server error` and it'd be hard to figure out what happened.

### CODE REVIEW

**Comment by eahefnawy** [comment]
> Do we understand what actually happens and why internal server error is returned?

Yeah because the user code is trying to load modules that don't exist, which is why we should validate it before publishing.

**Comment by eahefnawy** [comment]
Ok, I forgot to clarify that this only happens if you forgot to run `npm install` prior to publishing if you're using any packages. So it's expected and not a bug. Here we just give early feedback.

**Comment by medikoo** [comment]
> Yeah because the user code is trying to load modules that don't exist

Load in what way? Via `require` ?  Why it's an `internal server error` and not a meaningful error that's thrown?

**Comment by eahefnawy** [comment]
Yeah, we probably don't try/catch that deployment error. Nonetheless, it should be validated on the CLI during publishing, not during deploying, for the earliest feedback, hence this change.

**Comment by medikoo** [comment]
I'm still wondering why we expose it as "internal server error" (?) Invalid requires are reported  in lambdas  with "module initialization error: Error".

Doesn't it signal a bug in error handling on our side?

I understand that this patch will fix this single error case, still I think it'll be good to investigate it further, to fully understand why actually it is reported with meaningless "internal server error" message and why we do not have access to stack trace that'll give us valuable debugging information (?)

**Comment by eahefnawy** [comment]
> I'm still wondering why we expose it as "internal server error" (?) Invalid requires are reported in lambdas with "module initialization error: Error".

We don't expose an internal server error. It's probably APIG that does that because it's unhandled. We could handle it, but that's beyond the scope of this PR. I'm already tracking that separate issue here (which applies to other errors too):
https://github.com/serverlessinc/platform/issues/51

Meanwhile, let's merge this in until we figure this out. It provides better dx/feedback for this specific case regardless of how we end up handling different errors.

**Comment by medikoo** [comment]
> It's probably APIG that does that

ok, I missed that part.

> I'm already tracking that separate issue here (which applies to other errors too):
serverlessinc/platform#51

👍

File: src/cli/utils.js [hunk 1]
```diff
@@ -674,6 +674,32 @@ const isChinaUser = () => {
   return result;
 };
 
+/**
+ * Makes sure user ran "npm install" if package.json with dependencies was found
+ */
+const validateNodeModules = async (directory) => {
+  const srcPath = path.resolve(process.cwd(), directory);
```
Comment (medikoo) [inline src/cli/utils.js:681]: nit: with `path.resolve` paths are resolved naturally against `process.cwd()`, so that first argument can be skipped

**Pull request merged**